### PR TITLE
{ApiManagement} Fix #16868: Update description of ResourceId param 'New-AzApiManagementBackend' and 'Set-AzApiManagementBackend' cmdlet

### LINKED ISSUE #16868: New/Get-AzApiManagementBackend -ResourceId are inconsistent with each other
### Description

Failed to create a new apimanagementbackend when doing -resourceid like /susbcription/....
```
New-AzApiManagementBackend -Context  $apimContext -BackendId 123 -Url 'https://contoso.com/awesomeapi' -Protocol http -Title "first backend" -SkipCertificateChainValidation $true -Credential $credential -Description "my backend" -ResourceId /subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/resourceGroups/yufan-rg/providers/Microsoft.ApiManagement/service/contoso
```
It can be succeeded when doing -resourceid http://...
```
New-AzApiManagementBackend -Context  $apimContext -BackendId 123 -Url 'https://contoso.com/awesomeapi' -Protocol http -Title "first backend" -SkipCertificateChainValidation $true -Credential $credential -Description "my backend" -ResourceId https://management.azure.com/subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/resourceGroups/yufan-rg/providers/Microsoft.ApiManagement/service/contoso
```
The Get-AzApiManagementBackend can be used by doing both http://... and /susbcription/....
Also, the -resourceid won't check whether it's an Azure resource. It can be accepted by any url like -url parameter.

### Script or Debug output

```PowerShell
=== Request Data ===
Authority Provided? - True
Scopes - https://management.core.windows.net//.default
Extra Query Params Keys (space separated) -
ApiId - AcquireTokenSilent
IsConfidentialClient - False
SendX5C - False
LoginHint ? False
IsBrokerConfigured - False
HomeAccountId - False
CorrelationId - 32808c1b-fa75-4a76-9a78-6d4c16fcd149

DEBUG: (False) MSAL 4.30.1.0 MSAL.NetCore Microsoft Windows 10.0.19044 [01/13/2022 08:17:02 - ] === Token Acquisition (SilentRequest) started:

        Authority Host: login.microsoftonline.com
DEBUG: (False) MSAL 4.30.1.0 MSAL.NetCore Microsoft Windows 10.0.19044 [01/13/2022 08:17:02 - ] Azure region was not configured or could not be discovered. Not using a regional authority.
DEBUG: (False) MSAL 4.30.1.0 MSAL.NetCore Microsoft Windows 10.0.19044 [01/13/2022 08:17:02 - ] Access token is not expired. Returning the found cache entry. [Current time (01/13/2022 08:17:02) - Expiration Time (01/13/2022 09:28:45 +00:00) - Extended Expiration Time (01/13/2022 09:28:45 +00:00)]
DEBUG: (False) MSAL 4.30.1.0 MSAL.NetCore Microsoft Windows 10.0.19044 [01/13/2022 08:17:02 - ] Returning access token found in cache. RefreshOn exists ? False
DEBUG: (False) MSAL 4.30.1.0 MSAL.NetCore Microsoft Windows 10.0.19044 [01/13/2022 08:17:02 - ] Fetched access token from host login.microsoftonline.com.
DEBUG: (False) MSAL 4.30.1.0 MSAL.NetCore Microsoft Windows 10.0.19044 [01/13/2022 08:17:02 - ] === Token Acquisition finished successfully. An access token was returned with Expiration Time: 01/13/2022 09:28:45 +00:00 and Scopes https://management.core.windows.net//user_impersonation https://management.core.windows.net//.default
DEBUG: SharedTokenCacheCredential.GetToken succeeded. Scopes: [ https://management.core.windows.net//.default ] ParentRequestId:  ExpiresOn: 2022-01-13T09:28:45.0000000+00:00
DEBUG: [Common.Authentication]: Received token with LoginType 'User', Tenant: '54826b22-38d6-4fb2-bad9-b7b93a3e9c5a', UserId: 'v-yufanyang@microsoft.com'
DEBUG: ============================ HTTP REQUEST ============================

HTTP Method:
PUT

Absolute Uri:
https://management.azure.com/subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/resourceGroups/yufan-rg/providers/Microsoft.ApiManagement/service/yufanContosoApi/backends/123?api-version=2020-12-01

Headers:
x-ms-client-request-id        : 5e9f91c3-91d3-4299-8ba6-eef24fb162bf
Accept-Language               : en-US

Body:
{
  "properties": {
    "title": "first backend",
    "description": "my backend",
    "resourceId": "/subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/resourcegroups/yufan-rg/providers/Microsoft.Web/sites/yufanfunction",
    "credentials": {
      "query": {
        "sv": [
          "xx",
          "bb"
        ],
        "sr": [
          "cc"
        ]
      },
      "header": {
        "x-my-1": [
          "val1",
          "val2"
        ]
      },
      "authorization": {
        "scheme": "basic",
        "parameter": "opensesame"
      }
    },
    "tls": {
      "validateCertificateChain": false
    },
    "url": "https://contoso.com/awesomeapi",
    "protocol": "http"
  }
}


DEBUG: ============================ HTTP RESPONSE ============================

Status Code:
BadRequest

Headers:
Cache-Control                 : no-cache
Pragma                        : no-cache
Strict-Transport-Security     : max-age=31536000; includeSubDomains
X-Content-Type-Options        : nosniff
x-ms-request-id               : 16e140a6-9a15-4fc7-af38-997b26600ac4
Server                        : Microsoft-HTTPAPI/2.0
x-ms-ratelimit-remaining-subscription-writes: 1199
x-ms-correlation-request-id   : 4d554eb0-3c3a-4892-a74a-e668ee8c3a55
x-ms-routing-request-id       : SOUTHEASTASIA:20220113T081702Z:4d554eb0-3c3a-4892-a74a-e668ee8c3a55
Date                          : Thu, 13 Jan 2022 08:17:02 GMT

Body:
{
  "error": {
    "code": "ValidationError",
    "message": "One or more fields contain incorrect values:",
    "details": [
      {
        "code": "ValidationError",
        "target": "resourceId",
        "message": "Value should represent absolute http URL"
      }
    ]
  }
}


New-AzApiManagementBackend:
Error Code: ValidationError
Error Message: One or more fields contain incorrect values:
Request Id: 16e140a6-9a15-4fc7-af38-997b26600ac4
Error Details:
        [Code= ValidationError, Message= Value should represent absolute http URL, Target= resourceId]

DEBUG: AzureQoSEvent: Module: Az.ApiManagement:2.3.1; CommandName: New-AzApiManagementBackend; PSVersion: 7.2.0-preview.10; IsSuccess: False; Duration: 00:00:01.5593595; Exception:
Error Code: ValidationError
Error Message: One or more fields contain incorrect values:
Request Id: 16e140a6-9a15-4fc7-af38-997b26600ac4
Error Details:
        [Code= ValidationError, Message= Value should represent absolute http URL, Target= resourceId]
;
DEBUG: Finish sending metric.
DEBUG: 4:17:04 PM - NewAzureApiManagementBackend end processing.
```

### Environment data

```PowerShell
Name                           Value
----                           -----
PSVersion                      7.2.0-preview.10
PSEdition                      Core
GitCommitId                    7.2.0-preview.10
OS                             Microsoft Windows 10.0.19044
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Module versions

```PowerShell
ModuleType Version    PreRelease Name                                ExportedCommands
---------- -------    ---------- ----                                ----------------
Script     2.3.1                 Az.ApiManagement                    {Add-AzApiManagementApiToGateway, Add-AzApiManagementApiToProduct, Add-AzApiManagementProductToGroup, Add-AzApiManagementRegion…}
```

### Error output

```PowerShell
Message        :
                 Error Code: ValidationError
                 Error Message: One or more fields contain incorrect values:
                 Request Id: 56a0f409-f4f2-43e7-8fe3-dc388f2ec48a
                 Error Details:
                        [Code= ValidationError, Message= Value should represent absolute http URL, Target= resourceId]

StackTrace     :
Exception      : System.Exception
InvocationInfo : {New-AzApiManagementBackend}
Line           : New-AzApiManagementBackend -Context  $apimContext -BackendId 123 -Url 'https://contoso.com/awesomeapi' -Protocol http -Title "first backend" -SkipCertificateChainValidation $true -Credential $credential -Description
                 "my backend" -ResourceId /subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/resourceGroups/yufan-rg/providers/Microsoft.ApiManagement/service/contoso
Position       : At line:1 char:1
                 + New-AzApiManagementBackend -Context  $apimContext -BackendId 123 -Url …
                 + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
HistoryId      : 10
```

### PR DESCRIPTION
**Pull request by RakeshMohanMSFT**
## Checklist

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by RakeshMohanMSFT** [comment]
Fixes #16868

File: src/ApiManagement/ApiManagement/help/New-AzApiManagementBackend.md [hunk 1]
```diff
@@ -154,8 +154,7 @@ Accept wildcard characters: False
 
 ### -ResourceId
 Management Uri of the Resource in External System.
-This parameter is optional.
-This url can be the Arm Resource Id of Logic Apps, Function Apps or Api Apps.
```
Comment (msJinLei) [inline src/ApiManagement/ApiManagement/help/New-AzApiManagementBackend.md:158]: Please update the description in https://github.com/Azure/azure-powershell/blob/main/src/ApiManagement/ApiManagement.ServiceManagement/Commands/NewAzureApiManagementBackend.cs#L64

File: src/ApiManagement/ApiManagement/help/Set-AzApiManagementBackend.md [hunk 2]
```diff
@@ -197,8 +197,7 @@ Accept wildcard characters: False
 
 ### -ResourceId
 Management Uri of the Resource in External System.
-This parameter is optional.
-This url can be the Arm Resource Id of Logic Apps, Function Apps or Api Apps.
```
Comment (msJinLei) [inline src/ApiManagement/ApiManagement/help/Set-AzApiManagementBackend.md:201]: https://github.com/Azure/azure-powershell/blob/main/src/ApiManagement/ApiManagement.ServiceManagement/Commands/SetAzureApiManagementBackend.cs#L75

**Review by msJinLei — CHANGES_REQUESTED** [review]

File: src/ApiManagement/ApiManagement/ChangeLog.md [hunk 3]
```diff
@@ -19,6 +19,8 @@
 -->
 ## Upcoming Release
```
Comment (msJinLei) [inline src/ApiManagement/ApiManagement/ChangeLog.md:21]: Remove this empty line

**Review by msJinLei — CHANGES_REQUESTED** [review]

File: src/ApiManagement/ApiManagement/help/Set-AzApiManagementBackend.md (hunk 2 above)
Comment (RakeshMohanMSFT) [reply src/ApiManagement/ApiManagement/help/Set-AzApiManagementBackend.md:201]: Done

File: src/ApiManagement/ApiManagement/help/New-AzApiManagementBackend.md (hunk 1 above)
Comment (RakeshMohanMSFT) [reply src/ApiManagement/ApiManagement/help/New-AzApiManagementBackend.md:158]: Done

**Pull request closed without merging**